src/resources/env_config.rs

```diff
@@ -0,0 +1,233 @@
+}
+
+/// A third-party resource with nothing special to it.
+pub enum BasicResource {
+    UrlWithFallback {
+        url: String,
+        source: Option<String>
+    },
+    UrlOnly {
+        url: String
+    },
+    Omit
+}
+
+impl Resource for BasicResource {
+    fn must_render_url(&self) -> bool {
+        match self {
+            &BasicResource::UrlWithFallback{url: _, source: _} => true,
+            &BasicResource::UrlOnly{url: _} => true,
+            _ => false
+        }
+    }
+
+    fn must_embed(&self) -> bool {
+        match self {
+            &BasicResource::UrlWithFallback{url: _, source: _} => true,
+            _ => false
+        }
+    }
+
+    fn url(&self) -> String {
+        match self {
+            &BasicResource::UrlWithFallback{ref url, source: _} => url.clone(),
+            &BasicResource::UrlOnly{ref url} => url.clone(),
+            _ => panic!("no url available")
+        }
+    }
+
+    fn source(&self) -> Option<String> {
+        match self {
+            &BasicResource::UrlWithFallback{url: _, ref source} => source.clone(),
+            _ => panic!("no source available")
+        }
+    }
+}
+
+/// Special struct for font-awesome, as it consists of multiple embedded files
+/// but is configured as a directory. Methods implemented for this struct
+/// calculate the paths of all the files in this directory.
+pub struct Awesome {
+    resource: BasicResource
+}
+
+impl Resource for Awesome {
+    fn must_render_url(&self) -> bool {
+        self.resource.must_render_url()
+    }
+
+    fn must_embed(&self) -> bool {
+        self.resource.must_embed()
+    }
+
+    fn url(&self) -> String {
+        self.resource.url()
+    }
+
+    fn source(&self) -> Option<String> {
+        self.resource.source()
+    }
+}
+
+impl Awesome {
+    pub fn css_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/css/font-awesome.min.css")
+    }
+
+    pub fn eot_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.eot")
+    }
+
+    pub fn svg_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.svg")
+    }
+
+    pub fn ttf_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.ttf")
+    }
+
+    pub fn woff_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.woff")
+    }
+
+    pub fn woff2_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.woff2")
+    }
+
+    pub fn otf_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/FontAwesome.otf")
+    }
+}
+
+/// Configuration information for all the third-party resources. This specifies
+/// for every resource whether it should be included at all, and if so, just as
+/// an URL or also as an embedded resource.
+pub struct Configuration {
```

Comment:
Similar to `Awesome` and `BasicResource`, `Configuration` should also implement `Debug`. `Clone` is also a useful thing to have.